### ISSUE TITLE
generateNewWalletFile error

### ISSUE THREAD

**Issue by WilliamLann** [web3j/web3j#539]
> java.lang.ClassCastException: org.bouncycastle.jcajce.provider.asymmetric.ec.BCECPrivateKey cannot be cast to org.bouncycastle.jcajce.provider.asymmetric.ec.BCECPrivateKey
>     at org.web3j.crypto.ECKeyPair.create(ECKeyPair.java:52)
>     at org.web3j.crypto.Keys.createEcKeyPair(Keys.java:59)
>     at org.web3j.crypto.WalletUtils.generateNewWalletFile(WalletUtils.java:55)
get this error when run
> 		String wallet = WalletUtils.generateNewWalletFile(password, destinationDirectory, true);

If I restart tomcat7 it will be ok,and still get the same error sometime later,not sure why!!!

**Comment by fcorneli** [comment]
Relates to https://github.com/web3j/web3j/commit/e689a3d41f4aa22784df988006dd70b085fe416b
See
https://github.com/e-Contract/ethereum-resource-adapter/blob/master/ethereum-rar-demo-model/src/main/java/be/e_contract/ethereum/rar/demo/model/BouncyCastleLifecycleBean.java
for an example of how to manage the lifecycle of BouncyCastle outside web3j (in this example, it's managed by an EJB3 container).

**Comment by BipinPaudel** [comment]
I was having a same problem in maven based project with wildfly server. The problem was due to dependency mismatch between web3j plugins like abi,core and crypto.

This is how I changed my pom dependency.
`  
        <dependency>
            <groupId>org.web3j</groupId>
            <artifactId>core</artifactId>
            <version>3.5.0</version>
            <exclusions>
                <exclusion>
                    <groupId>org.web3j</groupId>
                    <artifactId>crypto</artifactId>
                </exclusion>
            </exclusions>
        </dependency>

        
        <dependency>
            <groupId>org.web3j</groupId>
            <artifactId>crypto</artifactId>
            <version>3.5.0</version>
            <exclusions>
                <exclusion>
                    <groupId>org.web3j</groupId>
                    <artifactId>utils</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
        <dependency>
            <groupId>org.web3j</groupId>
            <artifactId>utils</artifactId>
            <version>3.5.0</version>
            <exclusions>
                <exclusion>
                    <groupId>org.bouncycastle</groupId>
                    <artifactId>bcprov-jdk15on</artifactId>
                </exclusion>
            </exclusions>
        </dependency>

        
        <dependency>
            <groupId>org.bouncycastle</groupId>
            <artifactId>bcprov-jdk15on</artifactId>
            <version>1.56</version>
            <scope>provided</scope>
        </dependency>`

I hope this will solve the problem. Thanks

**Comment by DevAmanVyas** [comment]
Tried the dependency but getting lifeCycle error

**Comment by ONE0ONE4** [comment]
> > > java.lang.ClassCastException：org.bouncycastle.jcajce.provider.asymmetric.ec.BCECPrivateKey无法强制
> > > 转换为org.web3j.crypto.ECKeyPair.create 中的org.bouncycastle.jcajce.provider.asymmetric.ec.BCECPrivateKey（ECKeyPair.java ：52）
> > > org.web3j.crypto.Keys.createEcKeyPair（Keys.java:59）
> > > org.web3j.crypto.WalletUtils.generateNewWalletFile（WalletUtils.java:55）
> > > 运行时遇到此错误
> > > String wallet = WalletUtils.generateNewWalletFile（password ，destinationDirectory，true）;
> > 
> > 
> > 如果我重新启动tomcat7它会没问题，并且以后仍会得到相同的错误，不知道为什么！
> 
> @WilliamLann你的问题解决了吗？我现在有同样的情况，我想征求意见

update 

> > > java.lang.ClassCastException: org.bouncycastle.jcajce.provider.asymmetric.ec.BCECPrivateKey cannot be cast to org.bouncycastle.jcajce.provider.asymmetric.ec.BCECPrivateKey
> > > at org.web3j.crypto.ECKeyPair.create(ECKeyPair.java:52)
> > > at org.web3j.crypto.Keys.createEcKeyPair(Keys.java:59)
> > > at org.web3j.crypto.WalletUtils.generateNewWalletFile(WalletUtils.java:55)
> > > get this error when run
> > > String wallet = WalletUtils.generateNewWalletFile(password, destinationDirectory, true);
> > 
> > 
> > If I restart tomcat7 it will be ok,and still get the same error sometime later,not sure why!!!
> 
> @WilliamLann Has your problem been solved? I have the same situation now, I would like to ask for advice

upgrade web3j version to latest version 4.0.0 can resolve this problem

**Comment by ghareeb-falazi** [comment]
> I was having a same problem in maven based project with wildfly server. The problem was due to dependency mismatch between web3j plugins like abi,core and crypto.
> This is how I changed my pom dependency.
> 
> ```
>    `<dependency>
>         <groupId>org.web3j</groupId>
>         <artifactId>core</artifactId>
>         <version>3.5.0</version>
>         <exclusions>
>             <exclusion>
>                 <groupId>org.web3j</groupId>
>                 <artifactId>crypto</artifactId>
>             </exclusion>
>         </exclusions>
>     </dependency>
> ```
> ```
>     <dependency>
>         <groupId>org.web3j</groupId>
>         <artifactId>crypto</artifactId>
>         <version>3.5.0</version>
>         <exclusions>
>             <exclusion>
>                 <groupId>org.web3j</groupId>
>                 <artifactId>utils</artifactId>
>             </exclusion>
>         </exclusions>
>     </dependency>
>     <dependency>
>         <groupId>org.web3j</groupId>
>         <artifactId>utils</artifactId>
>         <version>3.5.0</version>
>         <exclusions>
>             <exclusion>
>                 <groupId>org.bouncycastle</groupId>
>                 <artifactId>bcprov-jdk15on</artifactId>
>             </exclusion>
>         </exclusions>
>     </dependency>
> 
>     
>     <dependency>
>         <groupId>org.bouncycastle</groupId>
>         <artifactId>bcprov-jdk15on</artifactId>
>         <version>1.56</version>
>         <scope>provided</scope>
>     </dependency>`
> ```
> I hope this will solve the problem. Thanks

This works for me instead of managing BouncyCastle lifecycle myself. (I think this way, even if Web3J creates the Provider instance itself, it will be of the same version the rest of the application uses)

**Comment by AlexandrouR** [comment]
Is this still an issue with the latest version of web3j ?

**Comment by folex** [comment]
Still experiencing this issue with web3j `4.2.0` & `4.3.0`:
```
[info] fluence.effects.swarm.SwarmClientIntegrationSpec *** ABORTED ***
[info]   java.lang.ClassCastException: org.bouncycastle.jcajce.provider.asymmetric.ec.BCECPrivateKey cannot be cast to org.bouncycastle.jcajce.provider.asymmetric.ec.BCECPrivateKey
[info]   at org.web3j.crypto.ECKeyPair.create(ECKeyPair.java:52)
[info]   at org.web3j.crypto.Keys.createEcKeyPair(Keys.java:76)
[info]   at org.web3j.crypto.Keys.createEcKeyPair(Keys.java:70)
```

**Comment by pengbs** [comment]
same issue

**Comment by folex** [comment]
Should be fixed by #991
